### ISSUE TITLE
XamlNodeList missing nodes when using XamlDeferringLoader

### ISSUE THREAD

**Issue by DevJohnC** [cwensley/Portable.Xaml#91]
As seen [in this issue](https://github.com/dmetzgar/corewf/issues/6#issuecomment-317568928) Portable.Xaml produces an incomplete `XamlNodeList` when using a `XamlDeferringLoader`.

Take the example markup:

```
<ItemContainer>
	<ItemContainer.ItemTemplate>
		<TextBlock Text="Hello World!" />
	</ItemContainer.ItemTemplate>
</ItemContainer>
```

and C# code:

```
	public class ItemContainer
	{
		[XamlDeferLoad(typeof(DeferredLoader), typeof(Template))]
		public Template ItemTemplate { get; set; } = new Template();

		public class Template
		{
			public XamlNodeList TemplateNodes { get; }

			public Template()
			{
			}

			public Template(XamlReader reader)
			{
				TemplateNodes = new XamlNodeList(reader.SchemaContext);
				XamlServices.Transform(reader, TemplateNodes.Writer);
			}
		}

		private class DeferredLoader : XamlDeferringLoader
		{
			public override object Load(XamlReader xamlReader, IServiceProvider serviceProvider)
			{
				return new Template(xamlReader);
			}

			public override XamlReader Save(object value, IServiceProvider serviceProvider)
			{
				return ((Template)value).TemplateNodes.GetReader();
			}
		}
	}
```

and you'll end up with a `XamlNodeList` that looks identical to the first image in the [linked issue above](https://github.com/dmetzgar/corewf/issues/6#issuecomment-317568928) which differs from System.Xaml.

**Comment by ewinnington** [comment]
Thanks for filing this! I was away from keyboard on holiday this last week.

**Comment by ewinnington** [comment]
I've attached a project in .net framework 4.6.1, replicating on your ItemContainer the behavior of 10 nodes in System.Xaml versus 2 nodes in Portable.Xaml. 

```
string xamlSystem =
@"<ItemContainerSystem xmlns=""clr-namespace:XAMLDeferred;assembly=XAMLDeferred"" 
                 xmlns:y=""http://schemas.microsoft.com/winfx/2006/xaml/presentation""
                 xmlns:x=""http://schemas.microsoft.com/winfx/2006/xaml"">
    <ItemContainerSystem.ItemTemplate>
        <y:TextBlock Text = ""Hello World!"" />
     </ItemContainerSystem.ItemTemplate>
 </ItemContainerSystem>";

            string xamlPortable =
@"<ItemContainerPortable xmlns=""clr-namespace:XAMLDeferred;assembly=XAMLDeferred"" 
                 xmlns:y=""http://schemas.microsoft.com/winfx/2006/xaml/presentation""
                 xmlns:x=""http://schemas.microsoft.com/winfx/2006/xaml"">
    <ItemContainerPortable.ItemTemplate>
        <y:TextBlock Text = ""Hello World!"" />
     </ItemContainerPortable.ItemTemplate>
 </ItemContainerPortable>";

var ObjectSystemXaml = System.Xaml.XamlServices.Load(GenerateStreamFromString(xamlSystem)) as ItemContainerSystem;
var ObjectPortableXaml = Portable.Xaml.XamlServices.Load(GenerateStreamFromString(xamlPortable)) as ItemContainerPortable;

Console.WriteLine("ObjectSystemXaml : Nodes : " + ObjectSystemXaml.ItemTemplate.TemplateNodes.Count);
Console.WriteLine("ItemContainerPortable : Nodes : " + ObjectPortableXaml.ItemTemplate.TemplateNodes.Count);
```

Which results in:
ObjectSystemXaml : Nodes : 10
ItemContainerPortable : Nodes : 2

Maybe that can help to narrow it down. This does look like the behavior of[ loading the Workflow Foundation activities from xaml](https://github.com/dmetzgar/corewf/issues/6) that is blocking me.

[XAMLDeferred.zip](https://github.com/cwensley/Portable.Xaml/files/1789287/XAMLDeferred.zip)

**Comment by cwensley** [comment]
Thanks @ewinnington and @DevJohnC, the examples are great!  I'll take a look!

**Comment by ewinnington** [comment]
A bump to see if we can unblock this? It would be very helpful for me if I could move forward on CoreWf.

**Comment by cwensley** [comment]
Thanks @ewinnington for the bump.  I'll try to find some time soon when I'm not passed out from dealing with my two small children. (;

**Comment by cwensley** [comment]
Closing this as I believe @yariker fixed it in #98.
